packages/shared/src/guards.ts

```diff
@@ -329,16 +330,16 @@ export function assertDatasetValue<D extends Dataset<ScalarShape | ArrayShape>>(
 }
 
 export function assertDataLength(
-  arr: NdArray<number[]> | number[] | undefined,
-  dataArray: NdArray<number[]> | number[],
+  arr: AnyNumArray | undefined,
+  dataArray: AnyNumArray,
   arrName: string
 ) {
   if (!arr) {
     return;
   }
 
-  const { length: arrLength } = toArray(arr);
-  const { length: dataLength } = toArray(dataArray);
+  const { length: arrLength } = getValues(arr);
+  const { length: dataLength } = getValues(dataArray);
```

Comment:
Renamed `toArray` to `getValues` to make it clear that we're not converting anything to an array -- we're just retrieving the underlying numeric array in case the input argument is an ndarray.